[Seow Xuan Yi] iP

saltedfishxx opened the pull request:
# Macho Duke
```
       ／ﾌﾌ 　　　　　　ム｀ヽ
     / ノ)　   A　A 　      ）   ヽ
     / ｜　　(´・ω・`)ノ⌒ゝ_ノ
   /　ﾉ⌒7⌒ヽー    く　  ＼　／
  丶＿ ノ ｡　　   ノ､     |/
    　　 `ヽ `ー-'_人`ーﾉ
    　　　 丶 ￣ _人'彡ﾉ
   　　　　  ﾉ　r'十ヽ/
```
Want to feel **productive** and **motivated**? **_Macho cat is here!_** 

## Features

- [x] Sets tasks with various options such as to-do, deadline, events
- [x] Quickly mark your tasks as done, (or undone if you did an oopsie!)
- [x] Too many tasks to track? Fear not! Keyword search is here to save the day!
- [ ] Sort by urgent tasks (Coming soon) 

## Installing and running
1. Download the latest release `duke.jar` from [here](https://github.com/saltedfishxx/ip/releases).
2. In the directory, open your terminal command tool, pointing to the directory that `duke.jar` is in, run the following command:
```
java -jar duke.jar
```
3. Add your tasks with `todo`, `deadline`, or `events`.
4. Let it manage your tasks for you 😉

If you Java programmer, you can use it to practice Java too. Here's the `main `method:

```java
public static void main(String[] args) throws ErrorStorageException {
        new Duke("/tasks.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

saltedfishxx committed:
Complete Level-0: Rename, Greet, Exit

saltedfishxx committed:
Level 1 Completed: Echo

saltedfishxx committed:
Complete Level-1: Echo

saltedfishxx committed:
Complete Level-2: Add, List

saltedfishxx committed:
Merge remote-tracking branch 'origin/master'

saltedfishxx committed:
Complete Level-3: Mark as Done

- Added level 3 requirements
- Modified code to recognise input as [command] [value]
- Added some error handling

saltedfishxx committed:
Complete Level-4 and Level-5: ToDos, Events, Deadlines, Handling Errors

saltedfishxx committed:
Complete Level-6: Delete

saltedfishxx committed:
Added comments, debug var to exclude certain text.

saltedfishxx committed:
Renaming and reformatting for code standardisation

saltedfishxx committed:
Complete Level-7: Save

saltedfishxx committed:
Merge branch 'branch-level-7'

saltedfishxx committed:
Complete base Level-8: DateTime format

saltedfishxx committed:
Merge branch 'branch-level-8'

saltedfishxx committed:
Complete A-MoreOOP, A-Packages

saltedfishxx committed:
Merge remote-tracking branch 'origin/add-gradle-support'

saltedfishxx committed:
Complete A-Gradle: Use Gradle

saltedfishxx committed:
Complete A-JUnit: Add JUnit tests

saltedfishxx committed:
Completed A-Jar: Release v0.1

saltedfishxx committed:
Completed A-JavaDoc: Commenting

saltedfishxx committed:
Merge branch 'branch-A-JavaDoc'

saltedfishxx committed:
Complete A-CodingStandard and A-Checkstyle

saltedfishxx committed:
Merge branch 'branch-A-CodingStandard'

saltedfishxx committed:
Complete Level-9: Find

saltedfishxx committed:
Merge branch 'branch-Level-9'

saltedfishxx committed:
Implement javaFX tutorial part 3

saltedfishxx committed:
Complete javaFx tutorial

saltedfishxx committed:
Clean up of GUI implementation

saltedfishxx committed:
Integrate command operations to GUI

kwangthiag commented on src/main/java/helpers/Parser.java:
```diff
@@ -0,0 +1,288 @@
+
+    }
+
+    private enum CommandsList {
```
The enum class can be put as a separate class.

kwangthiag commented on src/main/java/helpers/Ui.java:
```diff
@@ -0,0 +1,232 @@
+     * Method to print out the starting logo
+     */
+    public void showStartLogo() {
+        String logo = isDebug ? ""
```
I think the logo can be put as a constant as it never changes.

kwangthiag commented on src/main/java/helpers/Ui.java:
```diff
@@ -0,0 +1,232 @@
+     * Method to print ending logo when user quits application
+     */
+    public void quit() {
+        String endLogo = isDebug ? ""
```
I think the logo can be put as a constant as it never changes.

saltedfishxx committed:
Update exception handling to implement on GUI

kwangthiag reviewed (commented):
The code is very clean, keep up the good work!

kwangthiag commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,79 @@
+ * A Parent class of Tasks to create tasks.
+ */
+public class Task {
+    protected String taskDesc;
```
The full name taskDescription could be written out

saltedfishxx committed:
Complete Level-10: GUI

saltedfishxx committed:
Hot fix to Level-10: Added exit application logic

saltedfishxx committed:
Merge branch 'branch-level-10'

saltedfishxx committed:
Complete A-Assertions: Adding Assert into TaskList.java

Add assertion statements to ensure data integrity

In this commit, we have added assertion statements to the 'addTask' and 'removeTask' methods
of the 'TaskList' class. These assertions help us ensure that the data manipulation in these
methods is performed correctly, preventing unexpected issues.

- In 'addTask', we assert that the input 'task' is not null before adding it to the list.
  This prevents accidental addition of null tasks.

- In 'removeTask', we assert that the 'index' provided is within the valid range of indices
  in the task list. This safeguards against out-of-bounds removal attempts.

These assertions improve the robustness of our code and make it easier to catch and debug
potential issues during development.

saltedfishxx committed:
Merge pull request #1 from saltedfishxx/branch-A-Assertions

Complete A-Assertions: Adding Assert into TaskList.java

saltedfishxx committed:
Extract enum CommandsList into CommandEnum class

saltedfishxx committed:
Merge pull request #2 from saltedfishxx/branch-A-CodeQuality

Extract enum CommandsList into CommandEnum class

saltedfishxx committed:
Completed C-Sort: Added sort command by ascending or descending

saltedfishxx committed:
Merge branch 'C-Sort'

saltedfishxx committed:
Removed unused command enums

saltedfishxx committed:
Fixed issue raised on code quality feedback

saltedfishxx committed:
Updated user guide and README.md
